### PR TITLE
Fix bug where OpenCV does not link with FFMPEG

### PR DESCRIPTION
**Pull request by furbrain**
## Description

Currently when building `opencv`, it does not find `ffmpeg`, even though it has been built and installed. This means that KeyframeSelector does not run under linux (opencv automatically bundles its own version of ffmpeg under windows).

## Features list

- [X] Fix https://github.com/alicevision/Meshroom/issues/2285

## Implementation remarks

This PR simply lets the opencv cmake know where to look for various libraries (and fixes a minor bug in getting the base docker image)

### CODE REVIEW

File: src/cmake/Dependencies.cmake [hunk 1]
```diff
@@ -973,6 +973,7 @@ if(AV_BUILD_OPENCV)
         CONFIGURE_COMMAND 
             ${CMAKE_COMMAND} 
             ${CMAKE_CORE_BUILD_FLAGS}
+            -DCMAKE_PREFIX_PATH=${CMAKE_INSTALL_PREFIX}
```
Comment (fabiencastan) [inline src/cmake/Dependencies.cmake:976]: Maybe better to add a variable
`set(FFMPEG_CMAKE_FLAGS -DFFMPEG_ROOT=${CMAKE_INSTALL_PREFIX})`
And add it bellow as it's done for the other libs "${TIFF_CMAKE_FLAGS} ${PNG_CMAKE_FLAGS} ${JPEG_CMAKE_FLAGS} ${LIBRAW_CMAKE_FLAGS}"

**Comment by fabiencastan** [comment]
@furbrain Will you have time to make the adjustments proposed in the review?

**Comment by furbrain** [comment]
I've tested using the FFMPEG_ROOT option as suggested above - this does not seem to let the opencv cmake find the ffmpeg libraries - possibly because it's actually looking for libav[codec|format|device|etc] libraries etc.

I'm not very good with cmake (python is more my area of expertise) and honestly I've struggled to debug why it doesn't work. I can't see any major harm from telling cmake to look in the general installation prefix for libraries.

Phil

**Comment by solonovamax** [comment]
I cannot compile from source as cuda seems to really dislike me for whatever reason.

Would it be possible by any chance for you to pass me a tarball for the meshroom distribution built using this patch?

or even, just a tarball of the alicevision build, as hopefully I should just be able to delete the existing alicevision directory and replace it with this fixed build.

**Comment by c-ewing** [comment]
@furbrain Do you mind if i take this over as there hasn't been any progress in a little while? I believe I have a working solution that addresses some of @fabiencastan's concerns.

Basically OpenCV is *VERY* particular about where packages are installed, especially FFmpeg and it doesn't provide a way of specifying where FFmpeg is located. @fabiencastan's  fix will not work as there is no way of specifying (that i have been able to find) the FFmpeg install path. The way around this when compiling OpenCV from source is adding the FFmpeg install location to the prefix path as @furbrain did such that the  `find_package()` used in OpenCV's build also searches in the install prefix defined during the AliceVision build. 

Adding the line:
```cmake
set(FFMPEG_CMAKE_FLAGS -DCMAKE_PREFIX_PATH=${CMAKE_INSTALL_PREFIX};${CMAKE_PREFIX_PATH})
```
to the FFmpeg build section and then adding `{FFMPEG_CMAKE_FLAGS}` to the OpenCV section


This would change would the CMAKE_PREFIX_PATH for *just* OpenCV to include the  install path for FFmpeg. 

I've tested this using the Docker container and checked the output of `/opt/AliceVision_install/bin-deps/opencv_version -v` and verified that OpenCV is built against FFmpeg. 

Thanks,
cewing

**Comment by furbrain** [comment]
This is fine by me!

**Comment by fabiencastan** [comment]
https://github.com/alicevision/AliceVision/pull/1784

**Pull request closed without merging**